### PR TITLE
Feature/sort manifest files

### PR DESCRIPTION
**Pull request by johnnyaug**

### CODE REVIEW

**Comment by ozkatz** [comment]
@johnnyaug anyone reviewing these?

File: block/s3/inventory.go [hunk 1]
```diff
@@ -22,6 +23,9 @@ type manifest struct {
 	SourceBucket       string         `json:"sourceBucket"`
 	Files              []manifestFile `json:"files"`
 	Format             string         `json:"fileFormat"`
+	invBucket          string
```
Comment (nopcoder) [inline block/s3/inventory.go:26]: inventoryBucket

File: block/s3/inventory.go [hunk 2]
```diff
@@ -77,6 +81,36 @@ func (inv *Inventory) InventoryURL() string {
 	return inv.Manifest.URL
 }
 
+func readFileMetadata(ctx context.Context, logger logging.Logger, s3 s3iface.S3API, getParquetReader parquetReaderGetter, m *manifest) error {
```
Comment (nopcoder) [inline block/s3/inventory.go:84]: looks like it should be a method for manifest - populate and work on manifest

File: block/s3/inventory.go [hunk 3]
```diff
@@ -77,6 +81,36 @@ func (inv *Inventory) InventoryURL() string {
...
+			return err
+		}
+		m.firstKeyByFilename[filename] = ""
+		if len(rows) != 0 {
+			m.firstKeyByFilename[filename] = rows[0].Key
+		}
+		err = closeReader()
+		if err != nil {
+			logger.WithFields(logging.Fields{"bucket": m.invBucket, "key": filename}).
+				Error("failed to close parquet reader in readFileMetadata")
```
Comment (nopcoder) [inline block/s3/inventory.go:108]: usually I do no log the function name as part of the method - can have it as part of the key fields. but in this case I think our logger already capture the error location.
maybe "failed to close parquet while generate inventory"

File: block/s3/inventory.go [hunk 4]
```diff
@@ -38,35 +42,35 @@ type parquetReaderGetter func(ctx context.Context, svc s3iface.S3API, invBucket
...
+	sort.Slice(m.Files, func(i, j int) bool {
+		return m.firstKeyByFilename[m.Files[i].Key] < m.firstKeyByFilename[m.Files[j].Key]
+	})
+	return &Inventory{Manifest: m, S3: s3, getParquetReader: getParquetReader, logger: logger}, nil
 }
 
 type Inventory struct {
 	S3               s3iface.S3API
 	Manifest         *manifest
+	ctx              context.Context
```
Comment (nopcoder) [inline block/s3/inventory.go:67]: is this one used

Comment (johnnyaug) [reply block/s3/inventory.go:67]: yes, in inventory_iterator.py:81

**Pull request merged**